### DISCUSSION TITLE
Support json format with relevant functions

### DISCUSSION THREAD

**Discussion by ZhiHanZ** [databendlabs/databend#3911] [Ideas]
clickhouse json format is especially helpful for logging and monitoring application and we could support it
https://clickhouse.com/docs/en/interfaces/formats/#jsoneachrow
https://clickhouse.com/docs/en/sql-reference/functions/json-functions/

**Comment by flaneur2020** [comment]
this article have a few sentences about the JSON virtual columns feature in snowflake https://nattaylor.com/blog/2019/snowflake-internals/ :

> Created virtual columns for paths with JSON is amazing to me.
> You can get an idea of how they might calculate the stats for this here: [Listing Distinct Key Names/Common Schema in JSON Data](https://community.snowflake.com/s/article/listing-distinct-key-namescommon-schema-in-json-data)
> So you could build a list of the most common paths in a JSON object and then project those, and keep metadata statistics useful for joining, etc

**Comment by flaneur2020** [comment]
the author of snowflake internals speaks highly of the snowflake's loading JSON to a schemaless table feature: https://nattaylor.com/blog/2019/the-joy-of-snowflake/

> Getting started with Snowflake can be as simple as uploading data S3, configuring your bucket as a datasource, creating a **schemaless table**, loading the data – and then the truly joyful part – querying with flexibility and blazing speed.

**Reply by ZhiHanZ** [reply]
once we support json format, we could integrate with Vector for kubernetes logging and monitoring pipeline~

**Reply by bohutang** [reply]
https://docs.snowflake.com/en/sql-reference/data-types-semistructured.html This type we need?

**Reply by flaneur2020** [reply]
yep, not only support scanning json, but also same fast as a schema-ed table.

it seems snowflake could "guesses" a schema for the common struct from a JSON file.
